core/services/chainlink/application.go

```diff
@@ -123,7 +121,7 @@ type ChainlinkApplication struct {
 	SessionReaper            utils.SleeperTask
 	shutdownOnce             sync.Once
 	explorerClient           synchronization.ExplorerClient
-	subservices              []services.Service
+	subservices              []interface{} // services.Service or services.ServiceCtx
```

Comment:
Bleurgh this is not very pleasant. Perhaps we can make everything a services.ServiceCtx and just have it ignore the context if it doesn't need it?

Reply:
The problem here is that we will need to update CL node, terra and solana packages, right?
Alternatively, what if we introduce additional interface like this:
```
type StartWithContext interface {
    Start(ctx context.Context) error
}
```
and make affected services implementing both `Service` and `StartWithContext` interfaces.
Then we let/ old Start methods just panic.
When we call Start()s, we check if a `Service` instance implements `StartWithContext` as well and call the new version of `Start(ctx)` instead of `Start`.
Is this a good idea or not? 
Also, cc @jmank88